### PR TITLE
Added authentication router to ETOS API.

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#49

### Description of the Change

Authentication methods for Simple, LDAP and None added.
Simple = username/password authentication, validate against DB
LDAP = username/password authentication, validate against LDAP
None = No credentials

Authentication methods are required in order to request an API
token. This token is always required in order to communicate
with the API, even when using the "NoAuth" method.
To request a token, send a post request to /token with proper
credentials depending on the backend chosen.

A token will be returned which shall be added to the request
headers in the form of 'Authorization: Bearer TOKEN'
The token will expire after a while and a new token must be
requested at that time.

Note that the database is, at the moment, just a dictionary. A real database structure must be created later.

### Alternate Designs

There are multiple designs, this was easiest to implement and is still considered stable.

### Benefits

This design will bring security to APIs.

### Possible Drawbacks

More complexity for the users since a JWT must always be provided.
A database must be added to ETOS. Either use redis or another database.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

**Comment by t-persson** [comment]
Note that this will not be merged until documentation has been updated!

**Review by andjoe-axis — APPROVED** [review]
+1

File: tests/config.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "secret": "tests/dummysecret",
+    "issuer": "http://etos-api.com/token",
+    "authorization_backend": {
+        "name": "NoAuthentication",
```
Comment (fredjn) [inline tests/config.json:5]: We should documents somewhere that this is case sensitive, since it is linked directly to the name of the class.

**Review by fredjn — CHANGES_REQUESTED** [review]

File: tests/config.json (hunk 1 above)
Comment (t-persson) [reply tests/config.json:5]: Yes, I made a comment in this PR that the documentation would receive an update.

**Review by fredjn — APPROVED** [review]

**Review by fredjn — APPROVED** [review]

**Pull request closed without merging**